cli/flox-rust-sdk/src/models/pkgdb.rs

```diff
@@ -89,7 +93,32 @@ pub enum CallPkgDbError {
 ///
 /// Error JSON is parsed into a [CallPkgDbError::PkgDbError].
 pub fn call_pkgdb(mut pkgdb_cmd: Command) -> Result<Value, CallPkgDbError> {
+    // Configure pkgdb PATH with exact versions of everything it needs.
+    //
+    // Nix itself isn't pure, which is to say that it isn't built with a
+    // reference to `git` in its closure, so correspondingly depends upon
+    // finding it in $PATH to function. Funnily enough it is also not built
+    // to know where `nix` itself resides, and again relies on $PATH for that.
+    //
+    // This just isn't OK for us as we're looking for flox to operate reliably
+    // in "hostile" environments, which includes situation where a user may
+    // redefine or blat their $PATH variable entirely, so we always invoke
+    // pkgdb with an explicit PATH of our making.
+    let pkgdb_paths = [
+        Path::new(&*NIX_PKG_BIN),
```

Comment:
Id like to know where nix depends on nix.
All i can find is [`getSelfExe`](https://github.com/NixOS/nix/blob/538eb2617aab78f263dcfafff37784b9586ee7b9/src/libutil/current-process.cc#L113), where there are uses of it that fall back to "nix" (on the PATH) if `getSelfExe` returns an empty option.

Reply:
I don't recall exactly which invocations trigger the library calls which fail, but I do know that adding this fixed the problem that I encountered. We've covered a lot of ground with this issue, sorry I don't remember, but I'd be happy to try removing one or both to see what tests fail, if any.

Reply:
As mentioned below, I couldn't get it to fail in testing so updated the code to remove nix/bin from the PATH, and if it passes in CI I'll resolve the conversation.